src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/Target_ARM/ARMEmitter.cs

```diff
@@ -63,7 +124,55 @@ public void EmitJMP(ISymbolNode symbol)
         // bx reg
         public void EmitJMP(Register destination)
         {
-            Builder.EmitShort((short)(0x47 | ((byte)destination << 3)));
+            Builder.EmitShort((short)(((0x47) << 8) + ((byte)destination << 3)));
+        }
+
+        // mov pc, lr
+        public void EmitRET()
+        {
+            EmitMOV(TargetRegister.PC, TargetRegister.LR);
```

Comment:
`bx lr` is the official return instruction. It should be faster than `mov pc, lr` on recent hardware.